util/app/path/path.go

```diff
@@ -27,3 +28,56 @@ func Path(root, path string) (string, error) {
 	}
 	return appPath, nil
 }
+
+const outOfBoundsError string = "Out of bounds symlink found"
+
+// ContainsOutOfBoundsSymlinks determines if basePath contains any symlinks that
+// are absolute or point to a path outside of the basePath.
+func ContainsOutOfBoundsSymlinks(basePath string) (bool, string, error) {
+	found := ""
+	var err error
+	absBasePath, err := filepath.Abs(basePath)
+	if err != nil {
+		return false, "", err
+	}
+	err = filepath.Walk(absBasePath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.Mode()&os.ModeSymlink == os.ModeSymlink {
+			// We don't use filepath.EvalSymlinks because it fails without returning a path
+			// if the target doesn't exist.
+			target, err := os.Readlink(path)
+			if err != nil {
+				return err
+			}
+			repoPath, err := filepath.Rel(absBasePath, path)
+			if err != nil {
+				return err
+			}
+			if filepath.IsAbs(target) {
+				found = repoPath
+				return errors.New(outOfBoundsError)
+			}
+			dir := filepath.Dir(path)
+			joined := filepath.Join(dir, target)
+			rel, err := filepath.Rel(absBasePath, joined)
+			if err != nil {
+				return err
+			}
+			if strings.Contains(rel, "..") {
+				found = repoPath
+				// Return an error so we don't keep traversing the tree
+				return errors.New(outOfBoundsError)
+			}
+		}
+		return nil
+	})
+	if found != "" && err.Error() == outOfBoundsError {
```

Comment:
Instead of setting `found` separately, can we just embed it in the original error message? Something like this:

```go
return fmt.Errorf("encountered out of bounds symlink at %q: %w", found, outOfBoundsError)
```

Then you can use `errors.Is()` for the error check here?